packages/core/strapi/src/cli/commands/transfer/command.ts

```diff
@@ -45,14 +45,142 @@ const command = () => {
       .hook(
         'preAction',
         ifOptions(
-          (opts) => !(opts.from || opts.to) || (opts.from && opts.to),
+          (opts) =>
+            (opts.from && opts.to) || (opts.from && opts.toToken) || (opts.to && opts.fromToken),
           async () =>
-            exitWith(
-              1,
-              'Exactly one remote source (from) or destination (to) option must be provided'
-            )
+            exitWith(1, 'Only one remote source (from) or destination (to) option may be provided')
         )
       )
+      .hook('preAction', async (thisCommand) => {
+        const opts = thisCommand.opts();
+        const hasEnvUrl = process.env.STRAPI_TRANSFER_URL;
+        const hasEnvToken = process.env.STRAPI_TRANSFER_TOKEN;
+
+        const logDocumentation = () => {
+          console.info(
+            'ℹ️  Data transfer documentation: https://docs.strapi.io/dev-docs/data-management/transfer'
+          );
+        };
+
+        const logEnvironmentVariables = () => {
+          if (!hasEnvUrl && !hasEnvToken) {
+            console.info('ℹ️  No transfer configuration found in environment variables');
+            console.info(
+              '   → Add STRAPI_TRANSFER_URL and STRAPI_TRANSFER_TOKEN environment variables to make the transfer process faster for future runs'
+            );
+            return;
+          }
+
+          console.info('ℹ️  Found transfer configuration in your environment:');
+
+          if (hasEnvUrl) {
+            console.info(
+              `   → Environment STRAPI_TRANSFER_URL (${hasEnvUrl}) will be used as the transfer URL`
+            );
+          }
+
+          if (hasEnvToken) {
+            console.info(
+              '   → Environment STRAPI_TRANSFER_TOKEN value will be used as the transfer token'
+            );
+          }
+
+          console.info(); // Empty line for better readability
+        };
+
+        const determineDirection = async () => {
+          // If user has not provided a direction from CLI, log the documentation
+          if (!opts.from && !opts.to) {
+            logDocumentation();
+          }
+
+          logEnvironmentVariables();
+
+          if (opts.from) {
+            return opts.from;
+          }
+          if (opts.to) {
+            return opts.to;
+          }
```

Comment:
Shouldn't this be

```typescript
if (opts.from) {
  return 'from';
}
if (opts.to) {
  return 'to';
}
```

Right now if I do `strapi transfer --from [URL] --from-token [TOKEN]` it still asks me about a URL and token, and the direction message on line 134 also says 'send data to', because the `direction` will be set to my from URL.